src/main/java/org/openlmis/integration/dhis2/service/communication/ProcessedDataExchangeService.java

```diff
@@ -69,18 +78,46 @@ public void sendData(Schedule schedule) {
     final String periodEnum = dataset.getCronExpression();
     final int timeOffset = dataset.getTimeOffset();
 
+    Server server = schedule.getServer();
+
     Pair<ZonedDateTime, ZonedDateTime> periodRange;
     String formattedStartDate;
     if (sourceTable.equals("Requisition")) {
-      periodRange = periodGeneratorService.getLastRequisitionPeriod();
-      formattedStartDate = periodGeneratorService.formatDate(periodRange.getSecond(), "Monthly");
+      if (periodMappingId != null) {
+        PeriodMapping periodMapping = periodMappingRepository
+                .findById(periodMappingId)
+                .orElseThrow(() -> new NotFoundException(
+                        MessageKeys.ERROR_PERIOD_MAPPING_NOT_FOUND));
+
+        periodRange = periodGeneratorService.generateRange(periodMapping);
+        DhisPeriodType dhisPeriodType = dhisDataService
+                .getDhisPeriodTypes(server.getUrl(), server.getUsername(), server.getPassword())
+                .stream()
+                .filter(pt -> periodMapping.getDhisPeriod().equals(pt.getName()))
+                .findAny()
+                .orElseThrow(() -> new NotFoundException(MessageKeys.ERROR_PERIOD_TYPE_NOT_FOUND));
+
+        formattedStartDate = periodGeneratorService.formatDate(periodRange.getSecond(),
+                dhisPeriodType);
+      } else {
+        periodRange = periodGeneratorService.getLastRequisitionPeriod();
+        formattedStartDate = periodGeneratorService.formatDate(periodRange.getSecond(), "Monthly");
```

Comment:
please move "Monthly" to private static final variable

Reply:
I created a constant.